### PR TITLE
run0: various UI tweaks

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/run/run.c [hunk 1]
```diff
@@ -85,6 +85,7 @@ static char *arg_exec_path = NULL;
 static bool arg_ignore_failure = false;
 static char *arg_background = NULL;
 static sd_json_format_flags_t arg_json_format_flags = SD_JSON_FORMAT_OFF;
+static char *arg_shell_prompt_prefix = NULL;
```
Comment (YHNdnzj) [inline src/run/run.c:88]: Missing static destructor?

**Review by YHNdnzj — CHANGES_REQUESTED** [review]

File: src/run/run.c [hunk 2]
```diff
@@ -1356,6 +1381,24 @@ static int make_unit_name(sd_bus *bus, UnitType t, char **ret) {
         assert(t < _UNIT_TYPE_MAX);
         assert(ret);
 
+        /* Preferable use our PID + pidfd ID as identifier, if available. It's a boot time unique identifier
+         * managed by the kernel. Unfortunately only new kernels support this, hence we keep a lot of
+         * fallback logic in place. */
```
Comment (YHNdnzj) [inline src/run/run.c]: So I think we should utterly drop the whole dbus identifier stuff. It creates extra churn when soft-reboot is used and such, so why not always work with uuid when pidfd id is unavailable?

Comment (poettering) [reply src/run/run.c]: good point.

File: src/run/run.c [hunk 3]
```diff
@@ -1346,69 +1372,38 @@ static int transient_timer_set_properties(sd_bus_message *m) {
...
-        const char *unique, *id;
-        char *p;
         int r;
 
         assert(bus);
         assert(t >= 0);
         assert(t < _UNIT_TYPE_MAX);
         assert(ret);
 
-        r = sd_bus_get_unique_name(bus, &unique);
```
Comment (YHNdnzj) [inline src/run/run.c:1385]: It's no longer necessary to pass bus into this function then

File: src/run/run.c [hunk 4]
```diff
@@ -799,6 +804,17 @@ static int parse_argv_sudo_mode(int argc, char *argv[]) {
         assert(argc >= 0);
         assert(argv);
 
+        const char *e = secure_getenv("SYSTEMD_RUN_SHELL_PROMPT_PREFIX");
```
Comment (YHNdnzj) [inline src/run/run.c]: It's better to parse this from argv first, and fall back to this bit of logic only if we don't get anything?

Comment (YHNdnzj) [reply src/run/run.c]: All other opts follow that pattern too

File: src/run/run.c [hunk 5]
```diff
@@ -1346,69 +1372,38 @@ static int transient_timer_set_properties(sd_bus_message *m) {
...
         assert(ret);
 
-        r = sd_bus_get_unique_name(bus, &unique);
-        if (r < 0) {
-                sd_id128_t rnd;
+        /* Preferable use our PID + pidfd ID as identifier, if available. It's a boot time unique identifier
+         * managed by the kernel. Unfortunately only new kernels support this, hence we keep some fallback
+         * logic in place. */
+
+        PidRef self = PIDREF_NULL;
```
Comment (YHNdnzj) [inline src/run/run.c]: _cleanup_ is missing?

File: src/run/run.c [hunk 6]
```diff
@@ -1346,69 +1372,38 @@ static int transient_timer_set_properties(sd_bus_message *m) {
...
+        r = pidref_set_self(&self);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get my own PID: %m");
```
Comment (YHNdnzj) [inline src/run/run.c]: Maybe be specific: Failed to create reference to our own process?

**Comment by poettering** [comment]
updated, ptal.

**Review by YHNdnzj — APPROVED** [review]

File: src/run/run.c [hunk 7]
```diff
@@ -1375,70 +1375,38 @@ static int transient_timer_set_properties(sd_bus_message *m) {
...
 
-        assert(bus);
         assert(t >= 0);
         assert(t < _UNIT_TYPE_MAX);
         assert(ret);
 
-        r = sd_bus_get_unique_name(bus, &unique);
-        if (r < 0) {
-                sd_id128_t rnd;
+        /* Preferable use our PID + pidfd ID as identifier, if available. It's a boot time unique identifier
```
Comment (yuwata) [inline src/run/run.c]: Preferable -> Preferably ?

File: src/run/run.c [hunk 8]
```diff
@@ -1375,70 +1375,38 @@ static int transient_timer_set_properties(sd_bus_message *m) {
...
+        _cleanup_(pidref_done) PidRef self = PIDREF_NULL;
+        r = pidref_set_self(&self);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get reference to my own process: %m");
+
+        r = pidref_acquire_pidfd_id(&self);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to acquire pidfd ID of myself, defaulting to randomized unit name: %m");
 
+                /* We couldn't get the the pidfd id. In that case, just pick a random uuid as name */
```
Comment (yuwata) [inline src/run/run.c]: the the

**Comment by yuwata** [comment]
Please rebase

**Pull request merged**